### PR TITLE
updated query support doc

### PR DESCRIPTION
**Pull request by celinebrass**

### CODE REVIEW

File: docs/reference/query-support.md [hunk 1]
```diff
@@ -249,24 +256,27 @@ ReadySet supports the following data types:
...
 - `BIT`
-    - ReadySet will parse, but ignores the optional length field
+  - ReadySet will parse, but ignores the optional length field
 - `VARBIT`
-    - ReadySet will parse, but ignores the optional length field
+  - ReadySet will parse, but ignores the optional length field
 - `SERIAL`
 - `BIGSERIAL`
+- `ENUM`
+  - ReadySet support enums in MySql but not Postgres
```
Comment (nickelization) [inline docs/reference/query-support.md]: nit: MySQL (vs MySql). Also, typo of "support" vs "supports"

File: docs/reference/query-support.md [hunk 2]
```diff
@@ -83,48 +83,55 @@ select * from t1, t2 where t1.x = t1.y
...
+  - `ifnull`
+  - `month`
+  - `timediff`
+  - `addtime`
+  - `round`
+  - `json_typeof`
+  - `jsonb_typeof`
+  - `coalesce`
+  - `substr` and `substring`
+  - `greatest`
```
Comment (nickelization) [inline docs/reference/query-support.md]: We also support `least` as well as `greatest`

File: docs/reference/query-support.md [hunk 3]
```diff
@@ -249,24 +256,27 @@ ReadySet supports the following data types:
...
-    - ReadySet will parse, but ignores the optional length field
+  - ReadySet will parse, but ignores the optional length field
 - `VARBIT`
-    - ReadySet will parse, but ignores the optional length field
+  - ReadySet will parse, but ignores the optional length field
 - `SERIAL`
 - `BIGSERIAL`
+- `ENUM`
+  - ReadySet support enums in MySql but not Postgres
+- `CITEXT`
```
Comment (nickelization) [inline docs/reference/query-support.md:280]: Besides adding `ENUM` and `CITEXT` we may want to mention that we support array types now too (it's mentioned above that we support array literals, but there's nothing here about it being a supported type).

Comment (nickelization) [reply docs/reference/query-support.md:280]: (Although I'm a little fuzzy on the specific features/limitations we have in our array support -- I'm just going off of the fact that I know we support at least some array functionality.)

File: docs/reference/query-support.md [hunk 4]
```diff
@@ -83,48 +83,55 @@ select * from t1, t2 where t1.x = t1.y
...
 - Literal values
-    - String literals, quoted according to the SQL dialect being used (single quotes for postgresql, double or single quotes for mysql)
-        - ReadySet does not support string literals with charset or collation specifications
-    - Integer literals
-    - Float literals
-        - ReadySet does not support float literals using scientific (exponential) notation
-    - the `NULL` literal
-    - the `CURRENT_TIMESTAMP`, `CURRENT_DATE`, and `CURRENT_TIME` literals
-    - boolean literals `TRUE` and `FALSE`
+  - String literals, quoted according to the SQL dialect being used (single quotes for postgresql, double or single quotes for mysql)
```
Comment (nickelization) [inline docs/reference/query-support.md]: nit: as long as we're fixing up the formatting we may also want to change mysql to MySQL too

File: docs/reference/query-support.md [hunk 5]
```diff
@@ -83,48 +83,55 @@ select * from t1, t2 where t1.x = t1.y
...
+    - ReadySet does not support string literals with charset or collation specifications
+  - Integer literals
+  - Float literals
+    - ReadySet does not support float literals using scientific (exponential) notation
+  - the `NULL` literal
+  - the `CURRENT_TIMESTAMP`, `CURRENT_DATE`, and `CURRENT_TIME` literals
+  - boolean literals `TRUE` and `FALSE`
```
Comment (nickelization) [inline docs/reference/query-support.md]: Another nit, I know it was like this already and you just fixed the indentation, but as long as we're touching these lines, it feels a little sloppy that these three bullet points are not capitalized

File: docs/reference/query-support.md (hunk 3 above)
Comment (celinebrass) [reply docs/reference/query-support.md:280]: Yeah I also wasn't clear on it -- i actually have a list of a few things I would like Griffin to add when he comes back. I will add this to it.

File: docs/reference/query-support.md [hunk 6]
```diff
@@ -11,14 +11,14 @@ The easiest way to determine which queries are supported is to refer to the [Rea
 ReadySet supports the following clauses in SQL SELECT queries:
 
 - `SELECT` with a list of select expressions, all of which must be supported expressions (see “[Expressions](#expressions)”)
-    - ReadySet does not support scalar subqueries in the `SELECT` clause
+  - ReadySet does not support scalar subqueries in the `SELECT` clause
```
Comment (jseldess) [inline docs/reference/query-support.md]: This bullet is no longer a sub-bulled of the one before. We need to revert to the previous indentation. 

You can check this in the preview build: https://docs-git-cb-update-query-support-readyset.vercel.app/reference/query-support/

File: docs/reference/query-support.md [hunk 7]
```diff
@@ -11,14 +11,14 @@ The easiest way to determine which queries are supported is to refer to the [Rea
...
 - `DISTINCT`, modifying the select clause
 - `FROM`, with a list of tables (which may be implicitly joined), and (uncorrelated) subqueries
 - `JOIN` (see ["Joins"](#joins))
 - `WHERE`
-    - The `WHERE` clause must have a single condition [expression](#expressions)
+  - The `WHERE` clause must have a single condition [expression](#expressions)
```
Comment (jseldess) [inline docs/reference/query-support.md]: This bullet is no longer a sub-bulled of the one before. We need to revert to the previous indentation. 

You can check this in the preview build: https://docs-git-cb-update-query-support-readyset.vercel.app/reference/query-support/

File: docs/reference/query-support.md [hunk 8]
```diff
@@ -11,14 +11,14 @@ The easiest way to determine which queries are supported is to refer to the [Rea
...
 - `GROUP BY`, with a list of **column references**
-    - ReadySet does not support expressions or field positions in the `GROUP BY` clause
+  - ReadySet does not support expressions or field positions in the `GROUP BY` clause
```
Comment (jseldess) [inline docs/reference/query-support.md]: This bullet is no longer a sub-bulled of the one before. We need to revert to the previous indentation. 

You can check this in the preview build: https://docs-git-cb-update-query-support-readyset.vercel.app/reference/query-support/

File: docs/reference/query-support.md [hunk 9]
```diff
@@ -83,48 +83,56 @@ select * from t1, t2 where t1.x = t1.y
...
 - Literal values
-    - String literals, quoted according to the SQL dialect being used (single quotes for postgresql, double or single quotes for mysql)
-        - ReadySet does not support string literals with charset or collation specifications
-    - Integer literals
-    - Float literals
-        - ReadySet does not support float literals using scientific (exponential) notation
-    - the `NULL` literal
-    - the `CURRENT_TIMESTAMP`, `CURRENT_DATE`, and `CURRENT_TIME` literals
-    - boolean literals `TRUE` and `FALSE`
+  - String literals, quoted according to the SQL dialect being used (single quotes for PostgreSQL, double or single quotes for MySQL)
```
Comment (jseldess) [inline docs/reference/query-support.md]: Same as above. We need to revert to the previous indentation.

Same comment applied to various nested lists below as well. We should check throughout.

**Review by jseldess — CHANGES_REQUESTED** [review]

**Review by jseldess — APPROVED** [review]
LGTM! Thank you, @celinebrass! 

I know there are other updates still to add, but I'd like to get this PR merged and we can open follow-ups.

**Pull request merged**